### PR TITLE
Added Chat

### PR DESCRIPTION
**Pull request by RedonAdemi**

### CODE REVIEW

**Review by NathanAmanuelG — APPROVED** [review]

File: ReviveIT/Domain/Entities/Users.cs [hunk 1]
```diff
@@ -19,6 +21,10 @@ public class Users : IdentityUser
         public ICollection<UserCategory> UserCategories { get; set; }
         public ICollection<OperatingCity> OperatingCities { get; set; }
         public ICollection<PortfolioDocument> Portfolios { get; set; } = new List<PortfolioDocument>();
+        public ICollection<ChatSession> TechnicianChatSessions { get; set; }
+        public ICollection<ChatSession> CustomerChatSessions { get; set; }
+        public ICollection<ChatSession> CompanyChatSessions { get; set; } // Chat sessions where the user is the company
```
Comment (EdisonAlushaj) [inline ReviveIT/Domain/Entities/Users.cs:26]: Delete the comment in line 26

File: ReviveIT/Infrastructure/Data/ApplicationDbContext.cs [hunk 2]
```diff
@@ -25,6 +25,7 @@ public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : ba
         public DbSet<City> Cities { get; set; }
         public DbSet<OperatingCity> OperatingCities { get; set; }
         public DbSet<Feedback> Feedbacks { get; set; }
+
```
Comment (EdisonAlushaj) [inline ReviveIT/Infrastructure/Data/ApplicationDbContext.cs:28]: Delete the white line

File: ReviveIT/Infrastructure/Hubs/ChatHub.cs [hunk 3]
```diff
@@ -1,12 +1,76 @@
-﻿using Microsoft.AspNetCore.SignalR;
+﻿using Infrastructure.Data;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Logging;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Domain.Entities; // Ensure the namespace includes your entities
```
Comment (EdisonAlushaj) [inline ReviveIT/Infrastructure/Hubs/ChatHub.cs:6]: Delete the comment

File: ReviveIT/Infrastructure/Hubs/ChatHub.cs [hunk 4]
```diff
@@ -1,12 +1,76 @@
...
+            {
+                _logger.LogError("Message content is null or empty.");
+                return;
+            }
+
+            _logger.LogInformation($"Preparing to send message from {senderEmail} to {recipientId}");
+
+            try
+            {
+                // Log connection state before sending the message
```
Comment (EdisonAlushaj) [inline ReviveIT/Infrastructure/Hubs/ChatHub.cs:58]: Delete the comment

File: ReviveIT/Infrastructure/Hubs/ChatHub.cs [hunk 5]
```diff
@@ -1,12 +1,76 @@
...
+                _logger.LogInformation($"Connection state before sending: {Context.ConnectionId}, user: {senderEmail}");
+
+                // Attempt to send the message
```
Comment (EdisonAlushaj) [inline ReviveIT/Infrastructure/Hubs/ChatHub.cs:61]: Delete the comment

File: ReviveIT/Infrastructure/Hubs/ChatHub.cs [hunk 6]
```diff
@@ -1,12 +1,76 @@
...
+                await Clients.User(recipientId).SendAsync("ReceiveMessage", senderEmail, message);
+
+                // Log success
```
Comment (EdisonAlushaj) [inline ReviveIT/Infrastructure/Hubs/ChatHub.cs:64]: Delete the comment

File: ReviveIT/Infrastructure/Hubs/ChatHub.cs [hunk 7]
```diff
@@ -1,12 +1,76 @@
...
+                _logger.LogInformation("Message sent successfully.");
+            }
+            catch (Exception ex)
+            {
+                // Log any exceptions
```
Comment (EdisonAlushaj) [inline ReviveIT/Infrastructure/Hubs/ChatHub.cs:69]: Delete the comment

File: ReviveIT/Infrastructure/Hubs/ChatHub.cs [hunk 8]
```diff
@@ -1,12 +1,76 @@
...
+                _logger.LogError($"Error sending message from {senderEmail} to {recipientId}: {ex.Message}", ex);
+            }
         }
+
```
Comment (EdisonAlushaj) [inline ReviveIT/Infrastructure/Hubs/ChatHub.cs:73]: Delete white line

File: ReviveIT/Infrastructure/Hubs/ChatHub.cs [hunk 9]
```diff
@@ -1,12 +1,76 @@
...
     }
+
```
Comment (EdisonAlushaj) [inline ReviveIT/Infrastructure/Hubs/ChatHub.cs:75]: Delete white line

File: ReviveIT/WebUI/Views/Inbox/Inbox.cshtml [hunk 10]
```diff
@@ -1,49 +1,40 @@
...
-                    </div>
-                </div>
-            </a>
+            <input type="text" class="form-control my-3" placeholder="Search by email..." id="searchEmail">
+            <div id="userList">
+                <p>No chat sessions found.</p>
+            </div>
         </div>
 
+        <!-- Right column (Chat messages) -->
```
Comment (EdisonAlushaj) [inline ReviveIT/WebUI/Views/Inbox/Inbox.cshtml:31]: Delete comment

File: ReviveIT/WebUI/Views/Inbox/Inbox.cshtml [hunk 11]
```diff
@@ -53,34 +44,22 @@
...
-                    </div>
-                </div>
-            </div>
+            <div class="chat-messages" id="chat-messages"></div>
 
             <div class="chat-input">
                 <div class="input-group w-100">
-                    <input type="text" class="form-control" placeholder="Type a message...">
+                    <button class="btn btn-light" id="uploadButton">
+                        <i class="fas fa-image"></i> <!-- FontAwesome icon for image -->
```
Comment (EdisonAlushaj) [inline ReviveIT/WebUI/Views/Inbox/Inbox.cshtml:52]: Delete comment

File: ReviveIT/WebUI/Views/Inbox/Inbox.cshtml [hunk 12]
```diff
@@ -53,34 +44,22 @@
...
+                    </button>
+                    <input type="file" id="photoUpload" accept="image/*" style="display: none;">
+                    <input type="text" class="form-control" placeholder="Type a message..." id="chatMessage">
                     <div class="input-group-append">
-                        <button class="btn btn-primary">Send</button>
+                        <button class="btn btn-primary" id="sendButton">
+                            <i class="fas fa-paper-plane"></i> <!-- FontAwesome icon for send -->
```
Comment (EdisonAlushaj) [inline ReviveIT/WebUI/Views/Inbox/Inbox.cshtml:58]: Delete comment

**Review by EdisonAlushaj — CHANGES_REQUESTED** [review]

**Review by EdisonAlushaj — APPROVED** [review]

**Pull request merged**